lib/membrane_audio_mixer.ex

```diff
@@ -278,10 +307,20 @@ defmodule Membrane.AudioMixer do
       end)
       |> Enum.unzip()
 
-    payload = DoMix.mix(payloads, caps)
+    {payload, state} = mix_payloads(payloads, state)
     pads = Map.new(pads_list)
 
-    {payload, pads}
+    {payload, pads, state}
+  end
+
+  defp all_streams_ended?(%{pads: pads}) do
+    pads
+    |> Enum.map(fn {_pad, %{stream_ended: stream_ended}} -> stream_ended end)
+    |> Enum.all?()
+  end
+
+  defp last_preventer_wave?(%{prevent_clipping: prevent_clipping} = state) do
```

Comment:
It's still unclear what it does. First, the word `wave` should rather be replaced with something like `stream`, but I would do something different here - at first, I thought to call it `should_flush?`, but we can go even further and add a `flush` function to both mixers (see the previous comment) where one will return `{<<>>, state}` and the other result of `mix` with empty payload list